### PR TITLE
make p3 compatible

### PR DESCRIPTION
**Pull request by young-udemy**

### CODE REVIEW

**Comment by charlax** [comment]
Sure, thanks for the diff! Let me know when the tests pass.

**Comment by young-udemy** [comment]
Hello. Made further updates, and tests now pass.
Thanks.

File: multidimensional_urlencode/urlencoder.py [hunk 1]
```diff
@@ -23,7 +26,7 @@ def flatten(d):
         return [[d]]
 
     returned = []
-    for key, value in d.items():
+    for key, value in sorted(list(d.items())):
```
Comment (charlax) [inline multidimensional_urlencode/urlencoder.py]: Why sorted? Is that for the tests?

File: multidimensional_urlencode/urlencoder.py [hunk 2]
```diff
@@ -67,6 +70,6 @@ def urlencode(params):
         if isinstance(value, (list, tuple)):
             name += "[]"
 
-        url_params[name] = value
+        url_params.append((name, value))
```
Comment (charlax) [inline multidimensional_urlencode/urlencoder.py]: Why the change from dict to list?

Comment (young-udemy) [inline multidimensional_urlencode/urlencoder.py]: yes, for the tests to pass.
python2 returned consistently ordered list, but python3's `.item()` ordering is arbitrary.

[link 1](https://docs.python.org/3.0/whatsnew/3.0.html#views-and-iterators-instead-of-lists)
[link 2](http://stackoverflow.com/questions/14956313/dictionary-ordering-non-deterministic-in-python3)

File: multidimensional_urlencode/urlencoder.py (hunk 1 above)
Comment (charlax) [inline multidimensional_urlencode/urlencoder.py]: Actually, in both cases the order is undefined if I recall correctly. It seems a bit heavy handed to this sorting inside the library code, because that might be prohibitively expensive. What about just doing this formatting in tests? Or use `pprint`?

**Comment by young-udemy** [comment]
Hi, makes sense. reverted dict-to-list change, and updated tests instead.

**Comment by josiasmontag** [comment]
Could this be merged / version bumped? Would love Python 3 support. 
Thanks!

File: multidimensional_urlencode/urlencoder.py [hunk 3]
```diff
@@ -23,7 +26,7 @@ def flatten(d):
         return [[d]]
 
     returned = []
-    for key, value in d.items():
+    for key, value in list(d.items()):
```
Comment (charlax) [inline multidimensional_urlencode/urlencoder.py]: Why do we need this?

File: docs/index.rst [hunk 4]
```diff
@@ -19,10 +19,13 @@ multidimensional_urlencode lets you url encode a multidimensional dict:
...
 
-    >>> import urllib
+    >>> try:
+    ...     from urllib.parse import unquote
+    ... except:
+    ...     from urllib import unquote
     >>> from multidimensional_urlencode import urlencode
     >>> e = urlencode({"a": {"b": {"c": [1, 2, 3], "d": "e"}}})
-    >>> urllib.unquote(e)
+    >>> unquote(e)  # doctest: +SKIP
```
Comment (charlax) [inline docs/index.rst]: Why do we need to skip this?

File: README.rst [hunk 5]
```diff
@@ -13,10 +13,13 @@ multidimensional_urlencode lets you url encode a multidimensional dict:
...
 
-    >>> import urllib
+    >>> try:
+    ...     from urllib.parse import unquote
+    ... except:
+    ...     from urllib import unquote
     >>> from multidimensional_urlencode import urlencode
     >>> e = urlencode({"a": {"b": {"c": [1, 2, 3], "d": "e"}}})
-    >>> urllib.unquote(e)
+    >>> unquote(e)  # doctest: +SKIP
```
Comment (charlax) [inline README.rst]: Why do we need to skip this?

**Comment by charlax** [comment]
Sorry for the delay. I left a couple more comments, once they are reviewed we can merge this.

File: multidimensional_urlencode/urlencoder.py (hunk 3 above)
Comment (Grandmother) [inline multidimensional_urlencode/urlencoder.py]: We need this because python3 returns a view of dictionarie's items. [source](https://docs.python.org/3/library/stdtypes.html#dict.items)

File: README.rst (hunk 5 above)
Comment (charlax) [inline README.rst]: Gentle ping here - I'd like to know why we're skipping this test.

File: docs/index.rst (hunk 4 above)
Comment (charlax) [inline docs/index.rst]: Same.

File: multidimensional_urlencode/urlencoder.py (hunk 3 above)
Comment (charlax) [inline multidimensional_urlencode/urlencoder.py]: Yes for sure but it should still work as is...

**Comment by Grandmother** [comment]
Thanks a lot.
